src/model-helpers/User.js

```diff
@@ -0,0 +1,173 @@
+import {
+  buildThing,
+  createThing,
+  setThing,
+  saveSolidDatasetAt,
+  getSolidDataset,
+  createSolidDataset,
+  getThingAll,
+  getDatetime,
+  getPodUrlAll,
+  getStringNoLocale,
+  getUrl
+} from '@inrupt/solid-client';
+import { RDF_PREDICATES } from '../constants';
+
+import { setDocAclForUser } from '../utils/network/session-helper';
+
+/**
+ * @typedef {import('@inrupt/solid-ui-react').SessionContext} Session
+ */
+
+/**
+ * Function that updates a user's last active time on Solid Pod
+ *
+ * @memberof User
+ * @function updateUserActivity
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @param {URL} podUrl - the url of the user's pod to update
+ * @returns {Promise} Promise - Updates last active time of user to lastActive.ttl
+ */
+export const updateUserActivity = async (session, podUrl) => {
+  const activityDocUrl = `${podUrl}public/active.ttl`;
+  const accessObject = {
+    read: true,
+    append: true,
+    write: true,
+    control: true
+  };
+  const publicAccessObject = {
+    read: true,
+    append: false,
+    write: false,
+    control: false
+  };
+  let activityDataset;
+
+  const updateAndSaveActivity = async () => {
+    const activity = buildThing(createThing({ name: 'active' }))
+      .addDatetime(RDF_PREDICATES.dateModified, new Date())
+      .build();
+
+    activityDataset = setThing(activityDataset, activity);
+
+    await saveSolidDatasetAt(activityDocUrl, activityDataset, {
+      fetch: session.fetch
+    });
+  };
+
+  try {
+    activityDataset = await getSolidDataset(activityDocUrl, { fetch: session.fetch });
+    await updateAndSaveActivity();
+  } catch {
+    activityDataset = createSolidDataset();
+    await updateAndSaveActivity();
+
+    await setDocAclForUser(
+      session,
+      activityDocUrl,
+      'create',
+      session.info.webId,
+      accessObject,
+      publicAccessObject
+    );
+  }
+};
+
+/**
+ * Fetches the last time a user was active in pass
+ *
+ * @memberof User
+ * @function loadUserActivity
+ * @param {URL} podUrl - Url of user's pod to fetch from
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @returns {string} last time the user was active in PASS
+ */
+const loadUserActivity = async (podUrl, session) => {
+  const activityUrl = `${podUrl}public/active.ttl`;
+  try {
+    const solidDataset = await getSolidDataset(activityUrl, {
+      fetch: session.fetch
+    });
+    const activeThing = getThingAll(solidDataset)[0];
+    return getDatetime(activeThing, RDF_PREDICATES.dateModified);
+  } catch {
+    return null;
+  }
+};
+
+/**
+ * Creates a user object from a provided form submission
+ *
+ * @memberof User
+ * @function createUser
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @param {object} userSubmission - an object from a form submission containing the user creation data
+ * @returns {Promise} Promise - Updates last active time of user to lastActive.ttl
```

Comment:
We can use `{Promise<object>}` to update this JSDoc.

Think we could also update our typedefs for userListObject in typedefs.js so that it'll include podUrl.

<img width="628" alt="Screen Shot 2023-06-02 at 15 19 40" src="https://github.com/codeforpdx/PASS/assets/14917816/63a5a97e-bc21-4288-a36f-fd8765a2a1aa">

If we do that, we can use `{Promise<userListObject>}` instead. We should probably update the naming for types too with `UserListObject` instead, but think we can do that in another PR.